packages/build/src/plugins_core/blobs_upload/index.ts

```diff
@@ -0,0 +1,62 @@
+import { getDeployStore } from '@netlify/blobs'
+import pMap from 'p-map'
+
+import { addErrorInfo } from '../../error/info.js'
+import { log, logError } from '../../log/logger.js'
+
+import { getKeysToUpload, getBlobsDir, getFileWithMetadata, anyBlobsToUpload } from './utils.js'
+
+const coreStep = async function ({
+  debug,
+  logs,
+  deployId,
+  buildDir,
+  constants: { PUBLISH_DIR, SITE_ID, NETLIFY_API_TOKEN },
+}) {
+  const blobStore = getDeployStore({ siteID: SITE_ID, deployID: deployId, token: NETLIFY_API_TOKEN })
```

Comment:
If we supply an `apiURL` property here (constructed with the value of `constants.NETLIFY_API_HOST`) we can make this work on staging.

Reply:
Nice catch! Done in 06f58d8a9